Framework/Algorithms/test/CreateDetectorTableTest.h

```diff
@@ -86,6 +89,38 @@ class CreateDetectorTableTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove(ws->getName());
   }
 
+  void test_Exec_Matrix_Workspace_with_no_valid_spectra() {
+    Workspace2D_sptr inputWS = WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(1, 10);
+    auto &spec = inputWS->getSpectrum(0);
+    spec.clearDetectorIDs(); // clear the detectors to test the exception catching
+
+    CreateDetectorTable alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", std::dynamic_pointer_cast<MatrixWorkspace>(inputWS)));
+    TS_ASSERT_THROWS_NOTHING(alg.execute();)
```

Comment:
```suggestion
    TS_ASSERT_THROWS_NOTHING(alg.execute());
```
Did you mean this instead?

Reply:
I didn't spot that, yes I'd put the semicolons on the `TS_ASSERT`, etc statements as well, right at the end of the line.